ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/TrailingCommaRule.kt

```diff
@@ -192,6 +193,34 @@ public class TrailingCommaRule :
         node.reportAndCorrectTrailingCommaNodeBefore(inspectNode, emit, autoCorrect)
     }
 
+    private fun visitClassEntry(
+        node: ASTNode,
+        emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit,
+        autoCorrect: Boolean
+    ) {
+        val psi = node.psi
+        require(psi is KtClass)
+
+        if (psi.isEnum()) {
+            val classBody = node.children().singleOrNull { it.psi is KtClassBody }
+                ?: return // Nothing to do for empty enum class
+
+            val lastRelevantChild = classBody.psi.children.toList().asReversed()
+                .first()
```

Comment:
Please rewrite to
```
val lastRelevantChild = classBody.psi.lastChild
```
The word `Relevant` is confusing in this context. Why not just `lastChild`?

Reply:
I was trying to use this to find the semicolon element, it used to filter away the `ENUM_IGNORED_TYPES` but I never managed to get the semicolon since it seems to be removed from the AST

Reply:
Yeah, I think the semi colons are only needed to build the AST correctly. Please document this in a comment.

Reply:
I just bumped into test `Given an enumeration and the list of values is closed with a semicolon not followed by statements then do not return a lint error`. The semi colon after the enum must exist in the AST as in this test the unnecessary semicolon is reported.